### ISSUE TITLE
Migrating from gomz to aws-sdk-go?

### ISSUE THREAD

**Issue by echupriyanov** [emccode/rexray#323]
Will you consider switching library for communicating with AWS?
[aws-sdk-go](https://github.com/aws/aws-sdk-go) is officially supported by Amazon and closer to actual API.

I did preliminary rewrite, going to test it locally, and, if everything will work ok, submit it for review.

**Comment by akutz** [comment]
Hi @echupriyanov,

I have no issues with that. I know @clintonskitson originally forked goamz to add support for [copying snapshots](https://github.com/clintonskitson/goamz/tree/feature/snapcopy). Does AWS-SDK-Go support that? Also, how large in file size is the compiled AWS-SDK-Go archive file?

**Comment by akutz** [comment]
Hi @clintonskitson,

I just did a `go get` on this package and its sub-packages and sized it up. Wow. 

```
[0]akutz@pax:rexray$ du -m -d 1 /Users/akutz/Projects/go/pkg/darwin_amd64/github.com/aws
115	/Users/akutz/Projects/go/pkg/darwin_amd64/github.com/aws
```

@echupriyanov, are you aware to which packages we'd need to link or how incestuous the library is internally? If we link the root do we pull in the entire thing? I really don't want to add another 115MB to the size of the RR binary...

**Comment by clintonskitson** [comment]
@echupriyanov I would be open for the change if technically it made sense. There are some intricacies to how we pursue operations to minimize the amount of time waiting on operations. Goes back to the "eventually consistent" nature of the AWS API.

**Comment by echupriyanov** [comment]
Yes, AWS-SDK-Go has CopySnapshot. So far all operations are directly mapped to AWS API calls.
Right now unstripped linux binaries are as follows: Rexray (goamz) - ~68M, RexRay (aws-sdk-go) - ~74M.
Stripped binaries 38M and 41M corresponding.

**Comment by akutz** [comment]
Hi @echupriyanov,

So the size isn't a true factor. Then as @clintonskitson said, if it makes sense technically. For what reason are you suggesting it? What's the primary impetus to make the switch?

**Comment by echupriyanov** [comment]
Initially, I wanted to add feature to get AWS creds from AWS instance profile.
I’ve checked goamz, and it looks like its slightly outdated. Also, I’m using aws-sdk-go in my other projects, so I decided it would be a good practice to try to use it with RexRay.

At least we can try and look how it will be performing.

**Comment by echupriyanov** [comment]
I have finished porting RexRay to `aws-sdk-go`. I’ve tried all available operations and everything looks working (include cross-region snapshot copy).
Code is located here - https://github.com/echupriyanov/rexray/tree/aws-official-sdk

What do you think about it? Should I commit a PR?

And back to technical motivation for migration, I think, sticking to officially supported SDK can ease our life in the future, even if it makes not much difference right now.

**Comment by clintonskitson** [comment]
Yeah let's make it an official PR so we can track it that way.

**Comment by akutz** [comment]
Closed with PR #359
